From a59bc74f28168c62f23a251908ee96a7e29d4c80 Mon Sep 17 00:00:00 2001
From: Tomohiro Fujiwara <tomohiro.fujiwara.cw@hitachi.com>
Date: Tue, 5 Dec 2017 17:37:45 +0900
Subject: [PATCH 149/909] tee: Add a macro to provide a register function to a
 kernel space

commit aa1bb6d1927d617aea4afb021419326082653bf9 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

OP-TEE Driver is Added macros that provide the same function as
TEEC_RegisterSharedMemory in previous version.

Signed-off-by: Tomohiro Fujiwara <tomohiro.fujiwara.cw@hitachi.com>
Signed-off-by: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 include/linux/tee_drv.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/linux/tee_drv.h b/include/linux/tee_drv.h
index 392855b4ac1c..8d3b0ffec2ae 100644
--- a/include/linux/tee_drv.h
+++ b/include/linux/tee_drv.h
@@ -1,5 +1,6 @@
 /*
  * Copyright (c) 2015-2016, Linaro Limited
+ * Copyright (c) 2018, Renesas Electronics Corporation
  *
  * This software is licensed under the terms of the GNU General Public
  * License version 2, as published by the Free Software Foundation, and
@@ -33,6 +34,10 @@
 #define TEE_SHM_USER_MAPPED	BIT(4)  /* Memory mapped in user space */
 #define TEE_SHM_POOL		BIT(5)  /* Memory allocated from pool */
 
+/* Define functions corresponding to TEEC_RegisterSharedMemory */
+/* in kernel space with macros */
+#define tee_shm_reg tee_shm_alloc
+
 struct device;
 struct tee_device;
 struct tee_shm;
-- 
2.17.1

